mackup/main.py

```diff
@@ -38,35 +48,45 @@ def main():
     mckp = Mackup()
     app_db = ApplicationsDatabase()
 
+    def foreachApp(apps, f):
+        for app_name in apps:
+            if args.verbose: print(("{0} {1} {0}").format(header("---"), bold(app_name)))
+            f(app_name)
+            if args.verbose: print
+
     if args.mode == BACKUP_MODE:
         # Check the env where the command is being run
         mckp.check_for_usable_backup_env()
 
-        # Backup each application
-        for app_name in mckp.get_apps_to_backup():
-            app = ApplicationProfile(mckp, app_db.get_files(app_name))
-            app.backup()
+        def backupApplication(app_name):
```

Comment:
Instead if decaring a function inside a function that your are gonna pass to another function, it'd be easier if this logic was the responsibility of the ApplicationProfile class.